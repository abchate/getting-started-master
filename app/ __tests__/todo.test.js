const request = require('supertest');
const app = require('../src/index');
const db = require('../src/persistence'); // ajoute cette ligne

beforeAll(async () => {
    await db.init();
});

afterAll(async () => {
    await db.teardown();
});

describe('GET /items', () => {
    it('should return status 200 and an array', async () => {
        const res = await request(app).get('/items');
        expect(res.statusCode).toBe(200);
        expect(Array.isArray(res.body)).toBe(true);
    });

    it('should match the items snapshot', async () => {
        const res = await request(app).get('/items');
        expect(res.body).toMatchSnapshot();
    });
});
