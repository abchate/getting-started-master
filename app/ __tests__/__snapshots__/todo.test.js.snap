// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /items should match the items snapshot 1`] = `
[
  {
    "completed": false,
    "id": "7aef3d7c-d301-4846-8358-2a91ec9d6be3",
    "name": "Test",
  },
]
`;
